mpd/base.py

```diff
@@ -652,6 +692,7 @@ def connect(self, host, port=None, timeout=None):
         if IS_PYTHON2:
             self._rfile = self._sock.makefile("r")
             self._wfile = self._sock.makefile("w")
+            self._rbfile = self._sock.makefile("rb")
```

Comment:
I have not looked into the implementation of `makefile`. But since its buffered it looks like magic too me that using both `_sock.makefile("r")` and `_sock.makefile("rb")` does seamless work together.

Reply:
I agree, this looks somewhat magical. I'm doing tests right now and I'll see if I can fix this (probably by mostly-disabling buffering).